File: php/features.php
<?php
declare(strict_types=1);

// SouthParkQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SouthParkQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SouthParkQuotesBaseFeature();
            case "test":
                return new SouthParkQuotesTestFeature();
            default:
                return new SouthParkQuotesBaseFeature();
        }
    }
}

File: py/features.py
# SouthParkQuotes SDK feature factory

from feature.base_feature import SouthParkQuotesBaseFeature
from feature.test_feature import SouthParkQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SouthParkQuotesBaseFeature(),
        "test": lambda: SouthParkQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
